src/mscorlib/src/System/String.cs

```diff
@@ -973,17 +973,20 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
 
             if ((count == 0) || (omitEmptyEntries && this.Length == 0)) 
             {           
-                return new String[0];
+                return EmptyArray<String>.Value;
```

Comment:
I would make the change to return EmptyArray<String>.Value under #if FEATURE_CORECLR as @mburbea suggested. The change has likely marginal benefit - it is not worth discussing the compat risk on whether to take it everywhere.